Networking/NetworkingTests/Remote/ShippingLabelRemoteTests.swift

```diff
@@ -55,4 +55,45 @@ final class ShippingLabelRemoteTests: XCTestCase {
         XCTAssertEqual(printData.mimeType, "application/pdf")
         XCTAssertFalse(printData.base64Content.isEmpty)
     }
+
+    func test_refundShippingLabel_returns_refund_on_success() throws {
+        // Given
+        let orderID = Int64(279)
+        let shippingLabelID = Int64(134)
+        let remote = ShippingLabelRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "label/\(orderID)/\(shippingLabelID)/refund", filename: "shipping-label-refund-success")
+
+        // When
+        let result: Result<ShippingLabelRefund, Error> = try waitFor { promise in
+            remote.refundShippingLabel(siteID: self.sampleSiteID,
+                                       orderID: orderID,
+                                       shippingLabelID: shippingLabelID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        let refund = try XCTUnwrap(result.get())
+        XCTAssertEqual(refund, .init(dateRequested: Date(timeIntervalSince1970: 1607331363.627), status: .pending))
+    }
+
+    func test_refundShippingLabel_returns_error_on_failure() throws {
+        // Given
+        let orderID = Int64(279)
+        let shippingLabelID = Int64(134)
+        let remote = ShippingLabelRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "label/\(orderID)/\(shippingLabelID)/refund", filename: "shipping-label-refund-error")
+
+        // When
+        let result: Result<ShippingLabelRefund, Error> = try waitFor { promise in
+            remote.refundShippingLabel(siteID: self.sampleSiteID,
+                                       orderID: orderID,
+                                       shippingLabelID: shippingLabelID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        XCTAssertNotNil(result.failure)
```

Comment:
⚠️ Do you think it's worth it to test that we're also parsing the error? 

```
(lldb) po result.failure
▿ Optional<Error>
  ▿ some : Dotcom Error: [wcc_server_error_response] Error: The WooCommerce Shipping & Tax server returned: Bad Request Unable to request refund. The parcel has been shipped. ( 400 )
    ▿ unknown : 2 elements
      - code : "wcc_server_error_response"
      ▿ message : Optional<String>
        - some : "Error: The WooCommerce Shipping & Tax server returned: Bad Request Unable to request refund. The parcel has been shipped. ( 400 )"
```

I think this error is even useful to display to the user. I get this in WC Admin: 

![image](https://user-images.githubusercontent.com/198826/101413086-e7cfa500-38a0-11eb-9984-960b8b5596fb.png)

Reply:
Yes thanks for catching that! Updated in: https://github.com/woocommerce/woocommerce-ios/pull/3294/commits/a4ea4c4b5021a6027c6837ff61f00501544ce8cf

Reply:
> I think this error is even useful to display to the user. I get this in WC Admin:

Hmm, I'm surprised to see this message in wp-admin. I personally find the message not very user-friendly (the 400 error code and the "Error: The WooCommerce Shipping & Tax server returned" sound very technical and out of place) 🙄 I'll respond to your p2 comment and ask G about this!

Reply:
Yup. It's not user-friendly at all. Thanks for starting the discussion. 🙂